labelCloud/model/point_cloud.py

```diff
@@ -112,7 +145,35 @@ def from_file(
         points, colors = BasePointCloudHandler.get_handler(
             path.suffix
         ).read_point_cloud(path=path)
-        return cls(path, points, colors, init_translation, init_rotation, write_buffer)
+
+        labels = label_def = None
+        if cls.SEGMENTATION:
```

Comment:
If the logic differs to much, we could also use subclassing here → `SegmentationPointCloud`.

Yet on the other hand it might also be nice to show all "caught" points by a bounding box in the Bounding Box mode.

Reply:
Yes, I noticed this while I was implementing this. I considered to have a Mixin to plugin segmentation logic but as you pointed out, we might want to also have features that found points in the bounding box in the box mode. Agreed that we can do the overall look again before merging to master.